### ISSUE TITLE
Release process misses version bump on demo-build-config and lib-build-config

### ISSUE THREAD

**Issue by Clem69B** [eclipse/leshan#1297]
Since the version 2.0.0-M8, and the introduction of `build-config`, the release process misses version bump on the two sub-project `demo-build-config`, and `lib-build-config`.

From log of the last release jenkins job : https://ci.eclipse.org/leshan/job/leshan-release-java8/3/consoleFull
```
+ git add build-config/pom.xml leshan-bsserver-demo/pom.xml leshan-client-cf/pom.xml leshan-client-core/pom.xml leshan-client-demo/pom.xml leshan-core-cf/pom.xml leshan-core-demo/pom.xml leshan-core/pom.xml leshan-integration-tests/pom.xml leshan-server-cf/pom.xml leshan-server-core-demo/pom.xml leshan-server-core/pom.xml leshan-server-demo/pom.xml leshan-server-redis/pom.xml pom.xml
```
`build-config/demo-build-config/pom.xml` and `build-config/lib-build-config/pom.xml` are missing in git add command

To date, the project is not buildable without fix on `demo-build-config` and `lib-build-config` parent version.

**Comment by sbernard31** [comment]
@Clem69B, thx for reporting this :pray: 

**The issue :**

 If I understand the issue correctly, the commit : dc496e6aa97b236d43a46e5af5a5460f6ba46eac can not be built.

**The reason**:
Looking at jenkins config, I guess  : 
```
git add **/pom.xml pom.xml
```
does not add `demo-build-config`, and `lib-build-config` pom.xml file to the commit.

It seems to be because the shell used on jenkins is a sh-compatible one and globstar is no enable : 
```
$>  shopt
...
globstar       	off
...
```

(Locally it works with zsh on my machine but I can reproduce the issue with bash with globstar off)

**The plan :** 

I need to fix the jenkins config, I'm not sure what is the best way for now : 
- replace with something like : git add */pom.xml */*/pom.xml pom.xml
- try to activate globstar ?
- use anoher shell ?

And maybe I should also manually fix the dc496e6aa97b236d43a46e5af5a5460f6ba46eac commit as there is not so much commit after it ? (I mean by rewriting `master` branch)

**Comment by sbernard31** [comment]
Using : 
```
shopt -s globstar
git add **/pom.xml
shopt -u globstar
```
seems to do the job. 

See [temporary job](https://ci.eclipse.org/leshan/job/test-shell/5/console) that I used to test it. (It will be removed later)

--------------------------------

So I updated the [leshan-release-java8](https://ci.eclipse.org/leshan/job/leshan-release-java8/) job.
It should be fixed now. I will try to remember to double check this next release (2.0.0-M9).

I don't like to modify `master` history but as there is only few commits I think it worth it to change it to make **leshan-2.0.0-M8** tag buildable.  So I replaced buggy 04413db6ddfb29690e1e8365800fa89eaab4429a  commit by d80d7758ff8c08eed8cb2f2ac7e9b7cf86339277. I should work now.
